src/index.js

```diff
@@ -235,23 +234,39 @@ class CID {
    * @returns {void}
    */
   static validateCID (other) {
+    let errorMsg = CID._checkCIDComponents(other)
+    if (errorMsg) {
+      throw new Error(errorMsg)
+    }
+  }
+
+  /**
+   * Test if the given input is a valid CID object.
+   * Returns an error message if it is not, or
+   * Throws an error its multihash is invalid.
+   *
+   * @param {any} other
+   * @returns {string}
+   */
+  static _checkCIDComponents (other) {
```

Comment:
This function can be extracted to a separate file given it is an utility.

Reply:
I can create a file `cid-util.js` or `util.js` or `util/cid-util.js`. Any preferences?